src/basic_types.hpp

```diff
@@ -50,6 +52,45 @@ enum class BoundaryType : int {
   restricted
 };
 
+// Enumeration for accessing a field on different locations of the grid:
+// C = cell center of (i, j, k)
+// FX = x-face at (i - 1/2, j, k)
+// FY = y-face at (i, j - 1/2, k)
+// FZ = z-face at (i, j, k - 1/2)
+// EXY = edge at (i - 1/2, j - 1/2, k)
+// EXZ = edge at (i - 1/2, j, k - 1/2)
+// EXY = edge at (i, j - 1/2, k - 1/2)
+// NXYZ = edge at (i - 1/2, j - 1/2, k - 1/2)
+//
+// The values of the enumeration are chosen so we can do te % 3 to get
+// the correct index for each type of element in Variable::data
+enum class TopologicalElement : std::size_t {
+  C = 0,
+  FX = 3,
+  FY = 4,
+  FZ = 5,
+  EYZ = 6,
```

Comment:
This naming is consistent with the literature and perfectly clear -- however, I found that when using the names often, for my sanity I needed to define:
```
using TE = parthenon::TopologicalElement;
constexpr TE F1 = TE::FX;
constexpr TE F2 = TE::FY;
constexpr TE F3 = TE::FZ;
constexpr TE E1 = TE::EYZ;
constexpr TE E2 = TE::EXZ;
constexpr TE E3 = TE::EXY;
constexpr TE NN = TE::NXYZ;
```
I don't know if these would also be easier for other people, but I found it much nicer to look through e.g. emf calculations with numbers and consistent character widths, vs. remembering that `EXZ` is the one pointing in the Y direction, i.e. direction 2, so needs to be differenced in `j`, etc etc.

I wonder if you're attached to these names, or if we could support both schemes in user code?

Reply:
I am not attached to those names. I don't think we should support two naming schemes in Parthenon itself. That seems likely to be confusing for new users. Maybe `F1` instead of `FX` is more consistent with other things in the code, but I guess I prefer `X`, `Y`, and `Z` personally. I do like that your suggestions have the same character width.

Reply:
I don't have a strong opinion, but I did find the `X`, `Y`, `Z` notation a little hard to context switch to, given the rest of the code uses `1`, `2`, and `3`.

Reply:
Yeah, it probably makes sense to switch. X, Y, and Z are a little to tied to Cartesian coordinate systems, but I guess I just dislike the current convention since it naturally starts array indexing at 1.

Reply:
Just imagine 0 is reserved for time. :D

Reply:
How about aligning those with the conventions used for the coordinates https://parthenon-hpc-lab.github.io/parthenon/develop/src/coordinates.html
We discussed various options back then and it seems there was some agreement on the current ones.

Reply:
Fine by me.

Reply:
@pgrete: I think that makes sense. I guess to be consistent with that notation we would use what @bprather suggested (and also `CC` for cell-centered)?

Reply:
I am going to switch to 
```c++
enum class TopologicalElement : int { 
  CC = 0; 
  F1 = 3; 
  F2 = 4;
  F3 = 5;
  E1 = 6;
  E2 = 7;
  E3 = 8; 
  NN = 9; 
};
```
unless anyone has any other suggestions.

Reply:
done